gnucash/gnome-utils/gnc-gnome-utils.c

```diff
@@ -421,7 +421,39 @@ gnc_gnome_help (const char *file_name, const char *anchor)
     gboolean success;
 
     if (anchor)
-        uri = g_strconcat ("ghelp:", file_name, "?", anchor, NULL);
+    {
+        const gchar * const *sdatadirs = g_get_system_data_dirs ();
+        const gchar * const *langs = g_get_language_names ();
+        gchar *lookfor = g_strconcat ("gnome/help/", file_name, NULL);
+        gchar *help_path = NULL;
+        gchar *help_file = NULL;
+        gchar *out = NULL;
+
+        for (; *sdatadirs; sdatadirs++)
+        {
+            gchar *filename = g_build_filename (*sdatadirs, lookfor, NULL);
+            if (g_file_test (filename, G_FILE_TEST_EXISTS))
+                help_path = g_strdup (filename);
+            g_free (filename);
+        }
+
+        help_file = g_strconcat (file_name, ".xml", NULL);
+
+        for (; *langs; langs++)
+        {
+            gchar *filename = g_build_filename (help_path, *langs, help_file, NULL);
+            if (g_file_test (filename, G_FILE_TEST_EXISTS))
+                out = g_strdup (filename);
+            g_free (filename);
+        }
+
+        if (out)
+            uri = g_strconcat ("ghelp:", out, "?", anchor, NULL);
```

Comment:
How about making it smarter. Imagine anchor was added to C, but de got no update. Can we then fall back to C?

Reply:
Can not see an easy way of doing this.